Fix with_collection_parameter to be inherited from parent component (#1055)
github/view_component: docs/CHANGELOG.md

## main

> * Fix `with_collection_parameter` to be inherited from parent component

Reviewer: ```suggestion
* Fix bug where `with_collection_parameter` did not inherit from parent component.
```
